File: Structural_Patterns/Bridge/ExtendedAbstraction.cpp
#include "ExtendedAbstraction.hpp"
#include "Implementation.hpp"


ExtendedAbstraction::ExtendedAbstraction(Implementation* impl) : Abstraction(impl)
{
}

std::string ExtendedAbstraction::Operation() const
{
    return ("ExtendedAbstraction: Extended operation with: \n" + m_impl->OperationImplementation());
}

File: Structural_Patterns/Bridge/python/ExtendedAbstraction.py
# File name     : ExtendedAbstraction.py
# Description   : 
#
#


from Abstraction import Abstraction


class ExtendedAbstraction(Abstraction):
    '''
    You can extend the Abstraction without changing the Implementation classes.
    '''
    def operation(self) -> str:
        return (f"ExtendedAbstraction: Extended operation with:\n"
                f"{self.implementation.operationImpl()}")